lib/namebase/index.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const {
+  handleBatchOpen,
+  handleBatchReveal,
+  handleBatchRevealWithCache,
+  handleBatchBid,
+  handleBatchFinish,
+  handleSendMany,
+  handleBidWithCache,
+  handleOpenWithCache,
+  handleClearCache,
+  handleClearKeyFromCache
+} = require('./http');
+
+class NamebasePlugin {
+  constructor(node) {
+    this.node = node;
+    this.logger = node.logger.context('NamebasePlugin');
+  }
+
+  async open() {
+    const { node } = this;
+    this.logger.info('initializing namebase plugin');
+
+    if (!node.has('walletdb')) {
+      this.logger.error('walletdb plugin is not present, terminating '
+        + 'Namebase plugin execution');
+      return;
+    }
+
+    const walletPlugin = node.get('walletdb');
+    this.walletDB = walletPlugin.wdb;
+    this.extendWallet(walletPlugin);
+  }
+
+  async close() {
+    this.logger.info('terminating namebase plugin');
+  }
+
+  extendWallet(walletPlugin) {
+    this.logger.info('registering http routes');
+    const walletHttp = walletPlugin.http;
+    // set context
+    const context = {
+      walletDB: this.walletDB
+    };
+
+    // Endpoints
+    walletHttp.registerRoute('post', '/wallet/:id/openwithcache',
```

Comment:
This is new, what is the migration plan? How do we deploy this big change without downtime?